src/iterators.rs

```diff
@@ -1750,7 +1750,7 @@ impl PyHash for PathLengthMapping {
     }
 }
 
-impl<'py> PyEq<Bound<'py, PyAny>> for PathLengthMapping {
+impl PyEq<Bound<'_, PyAny>> for PathLengthMapping {
```

Comment:
I assume we needed that lifetime long ago but with a PyO3 upgrade it became redundant. I am glad clippy caught it.